api-tester/src/main/java/com/revenuecat/apitester/java/UpgradeInfoAPI.java

```diff
@@ -0,0 +1,31 @@
+package com.revenuecat.apitester.java;
+
+import com.android.billingclient.api.BillingFlowParams;
+import com.revenuecat.purchases.UpgradeInfo;
+
+@SuppressWarnings({"unused"})
+final class UpgradeInfoAPI {
+    static void check(final UpgradeInfo upgradeInfo) {
+        final String oldSku = upgradeInfo.getOldSku();
+        final String oldProductId = upgradeInfo.getOldProductId();
+        @BillingFlowParams.ProrationMode final Integer prorationMode = upgradeInfo.getProrationMode();
+
+        UpgradeInfo constructedUpgradeInfo = new UpgradeInfo(
+                upgradeInfo.getOldSku(),
+                upgradeInfo.getOldProductId(),
+                upgradeInfo.getProrationMode()
+        );
+
+        UpgradeInfo constructedUpgradeInfoNullProrationMode = new UpgradeInfo(
+                upgradeInfo.getOldProductId(),
+                upgradeInfo.getOldSku(),
+                null
+        );
+
+        UpgradeInfo constructedUpgradeInfoProductIdOnly = new UpgradeInfo(upgradeInfo.getOldProductId());
```

Comment:
i added calls to all the overloaded versions of the constructor. thought it might be overkill, but this actually made me realize we never @JvmOverloaded the constructor, so were requiring java users to pass in a proration mode.

Reply:
Yeah, I don't think it's overkill for the API tests, since it can actually help catching things like these 👍